src/solc_0.8/defi/rules/RequirementsRules.sol

```diff
@@ -0,0 +1,280 @@
+
+        // if it's a new member create a new registry, instead, only update
+        if (_islistERC1155Member(newContract) == false) {
+            _listERC1155Index.push(newContract);
+            _listERC1155[newContract].index = _listERC1155Index.length - 1;
+        }
+
+        emit ERC1155RequirementAdded(contractERC1155, ids, minAmountId, maxAmountId);
+    }
+
+    function getERC721ListRequirement(address contractERC721)
+        external
+        view
+        _isContract(contractERC721)
+        returns (RequirementRuleERC721 memory)
+    {
+        require(_islistERC721Member(IERC721(contractERC721)), "RequirementsRules: contract is not in the list");
+        return _listERC721[IERC721(contractERC721)];
+    }
+
+    function getERC1155ListRequirement(address contractERC1155)
+        external
+        view
+        _isContract(contractERC1155)
+        returns (RequirementRuleERC1155 memory)
+    {
+        require(_islistERC1155Member(IERC1155(contractERC1155)), "RequirementsRules: contract is not in the list");
+        return _listERC1155[IERC1155(contractERC1155)];
+    }
+
+    function deleteERC721fromListRequirement(IERC721 contractERC721)
+        external
+        onlyOwner
+        _isContract(address(contractERC721))
+    {
+        require(_islistERC721Member(contractERC721), "RequirementsRules: contract is not in the list");
+        uint256 indexToDelete = _listERC721[contractERC721].index;
+        IERC721 addrToMove = _listERC721Index[_listERC721Index.length - 1];
+        _listERC721Index[indexToDelete] = addrToMove;
+        _listERC721[addrToMove].index = indexToDelete;
+        _listERC721Index.pop();
+
+        emit ERC721RequirementDeleted(address(contractERC721));
+    }
+
+    function deleteERC1155fromListRequirement(IERC1155 contractERC1155)
+        external
+        onlyOwner
+        _isContract(address(contractERC1155))
+    {
+        require(_islistERC1155Member(contractERC1155), "RequirementsRules: contract is not in the list");
+        uint256 indexToDelete = _listERC1155[contractERC1155].index;
+        IERC1155 addrToMove = _listERC1155Index[_listERC1155Index.length - 1];
+        _listERC1155Index[indexToDelete] = addrToMove;
+        _listERC1155[addrToMove].index = indexToDelete;
+        _listERC1155Index.pop();
+
+        emit ERC1155RequirementDeleted(address(contractERC1155));
+    }
+
+    function _islistERC721Member(IERC721 reqContract) internal view returns (bool) {
+        if (_listERC721Index.length == 0) return false;
+
+        return (_listERC721Index[_listERC721[reqContract].index] == reqContract);
+    }
+
+    function _islistERC1155Member(IERC1155 reqContract) internal view returns (bool) {
+        if (_listERC1155Index.length == 0) return false;
+
+        return (_listERC1155Index[_listERC1155[reqContract].index] == reqContract);
+    }
+
+    function _checkERC721Requirements(address account) internal view returns (uint256) {
```

Comment:
why not public ?

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/commit/3eebb77ee852e2dd4fb5f5a788b3ed2ac64fb5bc